extensions/amp-date-countdown/1.0/date-countdown.js

```diff
@@ -0,0 +1,245 @@
+const DEFAULT_BIGGEST_UNIT = 'DAYS';
+const DELAY = 1000;
+
+/** @const {number} */
+const MILLISECONDS_IN_DAY = 24 * 60 * 60 * 1000;
+
+/** @const {number} */
+const MILLISECONDS_IN_HOUR = 60 * 60 * 1000;
+
+/** @const {number} */
+const MILLISECONDS_IN_MINUTE = 60 * 1000;
+
+/** @const {number} */
+const MILLISECONDS_IN_SECOND = 1000;
+
+const LOCALE_WORD = {
+  'de': ['Jahren', 'Monaten', 'Tagen', 'Stunden', 'Minuten', 'Sekunden'],
+  'en': ['Years', 'Months', 'Days', 'Hours', 'Minutes', 'Seconds'],
+  'es': ['años', 'meses', 'días', 'horas', 'minutos', 'segundos'],
+  'fr': ['ans', 'mois', 'jours', 'heures', 'minutes', 'secondes'],
+  'id': ['tahun', 'bulan', 'hari', 'jam', 'menit', 'detik'],
+  'it': ['anni', 'mesi', 'giorni', 'ore', 'minuti', 'secondi'],
+  'ja': ['年', 'ヶ月', '日', '時間', '分', '秒'],
+  'ko': ['년', '달', '일', '시간', '분', '초'],
+  'nl': ['jaar', 'maanden', 'dagen', 'uur', 'minuten', 'seconden'],
+  'pt': ['anos', 'meses', 'dias', 'horas', 'minutos', 'segundos'],
+  'ru': ['год', 'месяц', 'день', 'час', 'минута', 'секунда'],
+  'th': ['ปี', 'เดือน', 'วัน', 'ชั่วโมง', 'นาที', 'วินาที'],
+  'tr': ['yıl', 'ay', 'gün', 'saat', 'dakika', 'saniye'],
+  'vi': ['năm', 'tháng', 'ngày', 'giờ', 'phút', 'giây'],
+  'zh-cn': ['年', '月', '天', '小时', '分钟', '秒'],
+  'zh-tw': ['年', '月', '天', '小時', '分鐘', '秒'],
+};
+
+/**
+ * @param {!DateCountdownPropsDef} props
+ * @return {PreactDef.Renderable}
+ */
+export function DateCountdown({
+  endDate,
+  timeleftMs,
+  timestampMs,
+  timestampSeconds,
+  offsetSeconds = DEFAULT_OFFSET_SECONDS,
+  whenEnded = DEFAULT_WHEN_ENDED,
+  locale = DEFAULT_LOCALE,
+  biggestUnit = DEFAULT_BIGGEST_UNIT,
+  render,
+  children,
+}) {
+  useResourcesNotify();
+  const {playable} = useAmpContext();
+  const [epoch, setEpoch] = useState(
+    getEpoch(endDate, timeleftMs, timestampMs, timestampSeconds) +
+      offsetSeconds * DELAY
+  );
+  const [timeLeft, setTimeLeft] = useState(new Date(epoch) - new Date());
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      const newTimeLeft = new Date(epoch) - new Date();
+      setTimeLeft(() => newTimeLeft);
+      if (whenEnded === DEFAULT_WHEN_ENDED && newTimeLeft < 1000) {
+        clearInterval(interval);
+      }
+    }, DELAY);
+    return () => clearInterval(interval);
+  }, [playable]);
+
+  const data = Object.assign(
+    getYDHMSFromMs(timeLeft, biggestUnit),
+    getLocaleWord(locale)
+  );
+  return render(data, children);
+}
+
+/**
+ * Calculate the epoch time that this component should countdown to from
+ * one of multiple input options.
+ * @param {string|undefined} endDate
+ * @param {number|undefined} timeleftMs
+ * @param {number|undefined} timestampMs
+ * @param {number|undefined} timestampSeconds
+ * @return {number}
+ */
+function getEpoch(endDate, timeleftMs, timestampMs, timestampSeconds) {
+  let epoch;
+
+  if (endDate) {
+    epoch = Date.parse(endDate);
+  } else if (timeleftMs) {
+    epoch = Number(new Date()) + timeleftMs;
```

Comment:
`Number(new Date())` is simpler `Date.now()`.

Reply:
Thank you, updated!